litestar/middleware/session/base.py

```diff
@@ -145,6 +145,17 @@ def deserialize_data(data: Any) -> dict[str, Any]:
         """
         return cast("dict[str, Any]", decode_json(value=data))
 
+    @abstractmethod
+    def get_session_id(self, connection: ASGIConnection) -> str:
```

Comment:
```suggestion
    def get_session_id(self, connection: ASGIConnection) -> str | None:
```

Makes more sense than an arbitrary `null` string.

Reply:
How does `None` and `Empty` work together? All `SessionScope` attributes are `Empty` by default, yet for `session_id` we want to make it `str | None`. Should it then also be that type in `SessionScope`?

Otherwise, mypy will report type conflicts.

Reply:
The type in `SessionScope` should be `str | None | Empty` and then you could use the `litestar.utils.value_or_default` helper function that will get the value if it's not `Empty`, else it'll return a default value.